plugins/woocommerce/src/Utilities/NumberUtil.php

```diff
@@ -49,4 +53,85 @@ public static function array_sum( array $arr ): float {
 
 		return array_sum( $sanitized_array );
 	}
+
+	/**
+	 * Sanitize a cost value based on the current locale decimal and thousand separators.
+	 *
+	 * @param string $value               The value to sanitize.
+	 * @return string                     The sanitized value.
+	 * @throws \InvalidArgumentException If the value is not a valid numeric value.
+	 */
+	public static function sanitize_cost_in_current_locale( $value ): string {
+		$value                      = is_null( $value ) ? '' : $value;
+		$value                      = wp_kses_post( trim( wp_unslash( $value ) ) );
+		$currency_symbol_encoded    = get_woocommerce_currency_symbol();
+		$currency_symbol_variations = array( $currency_symbol_encoded, wp_kses_normalize_entities( $currency_symbol_encoded ), html_entity_decode( $currency_symbol_encoded ) );
+
+		$value = str_replace( $currency_symbol_variations, '', $value );
+
+		// Count the number of decimal points.
+		$decimal_point_count = substr_count( $value, '.' );
+
+		// If it's a standard decimal number (single decimal point and is_numeric), accept it directly. This could be in the case where the frontend has de-localised the value.
+		// We check for the decimal point count because is_numeric is looser than that.
```

Comment:
Should we make the comment more specific and explain how it's looser and the consequences that makes us do custom checks?

Reply:
Good callout because I forgot how it's looser by now 🤣 I'll update the comment

Reply:
```suggestion
		// We check for the decimal point count in addition to using is_numeric.
		// This is because is_numeric is much looser and accepts non-base10 numbers as well as 'e' to demarcate exponents.
```